### ISSUE TITLE
Prepare Release 3.8.0

### ISSUE THREAD

**Issue by hoffie** [jamulussoftware/jamulus#1595]
**Target timeline**
Scheduled feature freeze / Start of translation process: 2021-05-03
Targeted translation completion date: 2021-05-15
Approximate release date: 2021-06-02
Current state: Code freeze (rc) 

**Checklist**
Note: Please discuss potential wishes for changes in processes in the generic #1594 PR for this (future) issue template so that we can keep this issue focused on 3.8.0.

- [x] Assign this issue to the release shepherd who is in charge of managing this checklist
- [x] Create a milestone and add all relevant issues
- [x] Ensure that all issues/PR targeted for this release are done by checking the Project board with an appropriate filter (e.g. [`milestone:"Release 3.8.0"`](https://github.com/orgs/jamulussoftware/projects/2?card_filter_query=milestone%3A"release+3.8.0")). Reminder main developers to review entries in *Waiting on team* state. De-tag unfinished Issues/PRs.
- [x] Declare a code freeze by updating this Issue and adding a comment. PRs can still be worked on and may get reviewed, but must not be merged unless agreed explicitly.
- [x] Check ./Jamulus -h output against the en-Command-Line-Options.md wiki page and update if necessary. (@gilgongo, @pljones)
- [x] Write the Changelog (@hoffie) #1600
- [x] Start Website translations
  - [x] Create new branch `translate3_y_z` based on branch `release`
  - [x] Squash-merge `changes` into `translate3_y_z`
  - [x] Ping translators by creating an Issue for each language. Reminders for the Issue content: Note the deadline. Translators should open PRs with their changes against the `translate3_y_z` branch (?),  use a good commit message ("Add German translations for 3.y.z") and link to their respective Issue.
- [x] Prepare App translations
  - [x] Generate `.ts` files in master via `lupdate`
  - [x] Get the list of translators from src/utils.cpp (some grep/sed would be nice; last one per language is the active one).
  - [x] 3.8.0-specific: Tell translators about the Central->Directory change
  - [x] Create a translation Issue for each language, note the deadline, assign to the designated translator, add the Project, add the Milestone. Translators do their work using linguist and submit PRs with their changes against master. 
- [x] [Tag a beta release](https://github.com/jamulussoftware/jamulus/blob/master/RELEASE-PROCESS.md#steps-for-a-specific-release) (@softins)
- [x] Announce the beta release on Github Discussions. Pin the thread.
- [x] Get feedback on the stability of the beta release
- [x] Finish Website translations
  - [x] Wait for all PRs to be merged
  - [ ] Decide what to do about missing translations. Depending on the missing changes, consider keeping the old state (if it is still valid) or temporary backing out the relevant file(s) if they would become wrong.
  - [x] Merge `translate3_y_z` into `changes` via PR
- [ ] Finish App translations
  - [x] Review PRs <details><summary>Checklist for the PRs</summary> 
     ```
     - [ ] Translator listed in the `src/util.cpp` **optionally add link to PR or code**
     - [ ] Punctuation and spacing consistent
     - [ ] Signal words consistent ("ASIO", "Buffer")
     - [ ] App translations: No untranslated strings (`grep unfinished -5 src/res/translation/translation_$TRANSLATION*.ts`)
     - [ ] App translations: Only a single `.ts` file checked in
     - [ ] Installer translations: Passes `tools/check-wininstaller-translations.sh`
     ```
     </details>
  - [x] Wait for all PRs to be merged. If translations are missing after the targeted translation completion date, this is ok. They can be updated with the next release again.
  - [ ] Check for conflicting accelerator keys (see `tools/checkkeys.pl`)
  - [x] Generate `.qm` files via `lrelease Jamulus.pro`
- [x] Update Changelog again if necessary (3.8.0: clang-format)
- [x] [Tag a release candidate](https://github.com/jamulussoftware/jamulus/blob/master/RELEASE-PROCESS.md#steps-for-a-specific-release)
- [x] Announce the release candidate on Github Discussions. Pin the thread. Unpin and lock the beta thread.
- [x] Draft an announcement, include all contributors via `tools/get_release_contributors.py`
  - [x] Drop of pre-3.3.0 support
  - [x] Central -> Directory name change
- [x] [Update the version number in `Jamulus.pro` and add the release date to the Changelog header and commit](https://github.com/jamulussoftware/jamulus/blob/master/RELEASE-PROCESS.md#steps-for-a-specific-release)
- [x] Tag this commit as `r3_y_z`
- [x] [Force tag that tag as `latest` and push.](https://github.com/jamulussoftware/jamulus/blob/master/RELEASE-PROCESS.md#if-this-is-a-proper-release-move-the-latest-tag)
- [x] Wait for the build to complete
- [x] [Prepare `Jamulus.pro` (`dev` suffix) and ChangeLog (add a header) for the next release](https://github.com/jamulussoftware/jamulus/blob/master/RELEASE-PROCESS.md#make-the-master-branch-ready-for-post-release-development)
- [x] Do a smoke test for Windows/Mac/Linux -- Do the binaries start/connect properly?
- [x] Upload the artifacts to SourceForge and link latest and update the release announcement (a single sentence in the middle)
- [x] Trigger the update notification by updating the both Update Check Servers with the new version and update the release announcement (a single sentence in the middle)
- [x] Announce the new release with a summary of changes (+ link to the changelog for details) and a link to the download page
  - [x] On Github Discussions in the Announcements section. Lock the announcement thread. Pin the thread. Unpin and lock release candidate thread.
  - [x] On Facebook in the group "Jamulus (official group)". Turn off replies.
- [x] Update download links on the website by editing `config.yml`
- [x] Publish Website release by merging https://github.com/jamulussoftware/jamuluswebsite/pull/502
- [x] Check that all Issues and PRs tagged for this release are in Done/Closed state.
- [x] Create a milestone for the next minor release
- [x] Update [this checklist](https://github.com/jamulussoftware/jamulus/blob/master/.github/ISSUE_TEMPLATE/release_new_version.md) with any improvements which were found
- [x] Create a release retrospective discussion

**Comment by pljones** [comment]
> - [ ] Create a translation Issue for each language and assign to the designated translator. Translators do their work using linguist and submit PRs with their changes against master. Get the list of translators from src/utils.cpp (some grep/sed would be nice; last one per language is the active one). TODO: Tell translators about the Central->Directory name change.

Could this be broken out to:

- [ ] Get the list of translators from src/utils.cpp (some grep/sed would be nice; last one per language is the active one).
- [ ] Create a translation Issue for each language and assign to the designated translator.
- [ ] TODO: Tell translators about the Central->Directory name change.
- [ ] Translators do their work using linguist and submit PRs with their changes against master.

**Comment by pljones** [comment]
> Special things to mention: 
> - [ ] Ensure that Co-authors of commits are properly detected in tools/get_release_contributors.py (@hoffie)

That's not a special thing to mention, is it?  Hm.  Best alternative I can come up with is move the "Ensure" to the first entry and add the "Special things to mention" as a second entry with the two sub-items.  Doesn't really make any difference, though.

**Comment by pljones** [comment]
I don't see force tagging as latest? ("Tag this commit as r3_y_z" should be followed by "Force tag that tag as latest".  Or however Tony's doc explains it.)

**Comment by pljones** [comment]
> On Facebook in the group "Jamulus (official group)"

Also tag "turn off replies" or whatever Facebook call it.

**Comment by hoffie** [comment]
> That's not a special thing to mention, is it? Hm. Best alternative I can come up with is move the "Ensure" to the first entry and add the "Special things to mention" as a second entry with the two sub-items. Doesn't really make any difference, though.

It's a one time reminder for me to update the script because it does not handle that, iirc. In other words, Co-authors may be missing currently.

I'll integrate your other suggestions later. If you have more generic suggestions, could you post them to #1594 so that we can keep this Issue focused on the 3.8.0 specific parts?

**Comment by pljones** [comment]
:)

> Note: Please discuss potential wishes for changes in processes in the generic #1594 PR for this (future) issue template so that we can keep this issue focused on 3.8.0.

Put that at the top...

**Comment by hoffie** [comment]
I'm going to open tickets for translators this evening.

@jamulussoftware/maindevelopers We must not merge any PRs which touch translatable strings before the final by now. Necessary or planned (clang-format) code fixes are still possible.

**Comment by henkdegroot** [comment]
@hoffie , I checked the "Measurements" view in this release and noticed alignment "issue" for most languages (did not check them all). Should I open an issue for that so it can be addressed before a final release gets done?

**Comment by hoffie** [comment]
Yes, please do :)

**Comment by hoffie** [comment]
Regarding website translations, I have done the following:
- Created `translate3_8_0` based on `release`
- Squash-merged `changes` into that new branch. It now looks like this: https://github.com/jamulussoftware/jamuluswebsite/commit/8283940df62541b8313a801f5df5d334a88b440d

Can someone tell if that sounds and looks correct?
Can someone (@gilgongo?) come up with a text for the website translation Issue? I'm envisioning some instructions what to branch on etc. similar to the app translation Issues (e.g. https://github.com/jamulussoftware/jamulus/issues/1610).
Does someone have a list of website translators? If not, we'll have to work that out based on 3.7.0 git/PR history.
I'm planning to extend the translation issue helper tool (https://github.com/jamulussoftware/jamulus/pull/1594/files#diff-bbf341a6e62dcd8aec2f164cc7fe35bac0ac7ae4f0fee91967ff10babbc6554f) to cover website translations as well. I'll hopefully get around tomorrow evening.

**Comment by gilgongo** [comment]
Thanks! The changes on that branch look OK to me. I'm afraid I don't know who the web site translators were last time (and we may need to look at the last several releases to work it out as not all languages made it for 3.7).

As to the wording of the issue, based on yours it might be:

---

Update [LANG] website translation for 3.8.0

Hi!

We are getting ready for the 3.8.0 release and have created the "[translate_3.8.0](https://github.com/jamulussoftware/jamuluswebsite/commit/8283940df62541b8313a801f5df5d334a88b440d)" branch.

Because the English is still going through large changes, we are not expecting translations of anything other than the following pages:

- Home page
- [Setup](https://github.com/jamulussoftware/jamuluswebsite/blob/changes/wiki/en/en-Setup.md)/Getting Started
- Installation pages for each platform ([Win](https://github.com/jamulussoftware/jamuluswebsite/blob/changes/wiki/en/en-Installation-for-Windows.md), [Mac](https://github.com/jamulussoftware/jamuluswebsite/blob/changes/wiki/en/en-Installation-for-Macintosh.md), [Lin](https://github.com/jamulussoftware/jamuluswebsite/blob/changes/wiki/en/en-Installation-for-Linux.md))

While you are of course welcome to translate more than this, please be aware that you may have to re-translate for the next release.

We would be happy if you updated the translations for [LANG] until 2021-05-15.

Please

Update your fork to current master
Merge your translations to the translate_3.8.0 branch with the subject "Update [LANG] translation for 3.8.0"
Link your PR(s) to this issue so that we can track progress `not sure if non-admins can do that though?`

Feel free to use this Issue to discuss anything you need  prior to making any PR (and in your native language if needed).

Thanks for contributing to Jamulus!

---

**Comment by pljones** [comment]
I'm in a picky mood...

> Update your fork to current master

Will _everyone_ understand that means to jamulussoftware/jamuluswebsite master, rather than their own master?

> Merge your translations to the translate_3.8.0 branch with the subject "Update [LANG] translation for 3.8.0"

Should be 
> Branch from and merge to the translate_3.8.0 branch, raising a pull request with "Update [LANG] translation for 3.8.0"

They can put the issue number in the body of the pull request: https://github.com/jamulussoftware/jamulus/issues/1595 - I think that gets a mention in this ticket, then.

**Comment by hoffie** [comment]
We had some more iterations on this draft via chat, so the live version already covers some of these aspects, see https://github.com/jamulussoftware/jamuluswebsite/issues/456

> Will everyone understand that means to jamulussoftware/jamuluswebsite master, rather than their own master?

That had been clarified already.

> They can put the issue number in the body of the pull request: #1595 - I think that gets a mention in the ticket, then.

Yes, this had been added as well.

I'll rework this based on your comments some more (especially regarding "Branch from" as you suggested).

**Comment by hoffie** [comment]
I have updated the template in the issue creation tool and had it update the website translation issue texts. They also include a link to TRANSLATING.md now (thanks to @jujudusud for the pointer :)).

**Comment by dzpex** [comment]
@hoffie @pljones  @gilgongo  can you confirm that the It trans. is ok?

**Comment by hoffie** [comment]
> @hoffie @pljones @gilgongo can you confirm that the It trans. is ok?

@dzpex Yes, #1620 was merged already. Thanks! :)

FYI: There have been two last minute changes after 3.8.0beta1 (i.e. after we started translations). This required us to make some minor adjustments in all translation files which we assume to be (almost) correct (or at least understandable). It also added a few new "What's this" translations.

@jujudusud @henkdegroot already noticed that and submitted additional PRs for verifying and fixing this. We can still do that for other languages as well. We did not want to explicitly run another translation cycle because the potential inaccuracies should be really minor, because we do not want to delay the release and because we do not want to cause additional work by this.

To be clear, for all translators:
- It is OK not to submit another PR for 3.8.0.
- If you want to, you can have another look at the current code in `master` and submit any changes if required by Friday.

**Comment by hoffie** [comment]
Update: We're likely not releasing tomorrow because a problem on Mac has been discovered which is still being chased down (#1643).

**Comment by hoffie** [comment]
What are our options for moving forward with the release if we do not find a short-term solution for #1643?
1. Reverting the tab layout is not easily possible as lots of other PRs are based on it.
1. We could release despite this bug and advise Mac users to close the settings dialog if not needed
1. We could release despite this bug and add logic to warn Mac users if the settings dialog is kept open more than X seconds.
1. We could release despite this bug and start with metric updates disabled (on Mac) and only enable them after checking "Yes, I know this may cause issues on Macs" checkbox.
1. We could further delay the release.

**Comment by pljones** [comment]
I'd rather not _add_ more code for this, so either 2 or 5 seem the choices, with 2 being preferable (as it's "not our bug" :) ).  5 causes the problem that we'll potentially have the decision still waiting in a week if no solution is found.

**Comment by ann0see** [comment]
Although the website is out of sync and some translations are not up to date, I think we should release it at this state and update it after we have a release. Trying to fix stuff now will overly complicate things and might lead to even more confusion.

**Comment by gene96817** [comment]
I am impressed with the effort to provide all the translations. It is a big effort. Would it help to have something on the web pages to indicate if the translation is up to date? In this way, a reader might choose to struggle with reading another page if it matches the latest translation.

**Comment by gilgongo** [comment]
@gene96817 Under the current translation system, that would certainly be a good idea I think. But now we intend to move to p40a, we'll have the automatic "threshold" system which will prevent people being shown very out of date pages. It still might still lead to showing significantly wrong things in some cases (eg if a small but important bit f information has changed), but compared to how it works today it'll be a much better UX in most cases I think.

**Comment by gene96817** [comment]
Thanks for the reply and explanation. Also, thanks to everyone for all the improvements.

**Comment by ann0see** [comment]
Updated the download links for the website now

**Comment by hoffie** [comment]
As 3.8.0 is out of the door, I'm closing this. Thanks everyone!
